### ISSUE TITLE
Why TRTIS Accelerated limited?

### ISSUE THREAD

**Issue by ivankxt** [NVIDIA/tensorrt-inference-server#194]
I have successfully run the inception-plan model in TRTIS(Inference Server Release 18.08-py2, cuda9+cudnn7+TensorRT4 + GPU M40), and my model_repository is as below
plan_inception/
├── 1
│   └── model.plan
├── config.pbtxt
└── inception_labels.txt
we get the result as:
batch size = 100, and result time is 0.58s

As a comparison，I run the inception-graphdef model in  the same TRTIS, and my model_repository is as below
inception_graphdef/
├── 1
│   └── model.graphdef
├── config.pbtxt
└── inception_labels.txt
we get the result as:
batch size = 100, and result time is 0.78s

From which, we can know the inception-plan model in TRTIS accelerated limited. 
Could you give me some advices for that? Thank you.

**Comment by deadeyegoodwin** [comment]
Are you saying that you think you should have a much bigger speedup from using TensorRT plan compared with a TensorFlow graphdef? 

The speedup provided by TensorRT varies based on the model and how you use TensorRT. For example the M40 is a fairly old card and so TensorRT doesn't necessarily optimize it as well as more recent cards; specifically the M40 doesn't have TensorCores and so TensorRT can use mixed-precision optimizations, which tend to give a large speedup.

Also, the TensorFlow model is using the GPU, so you are seeing acceleration of both models relative to CPU.

Finally, you should try a more recent version of TRTIS and TensorRT as both have improved significantly since 18.08. 19.03 was just released so that would be a good version to try.

**Comment by ivankxt** [comment]
> Are you saying that you think you should have a much bigger speedup from using TensorRT plan compared with a TensorFlow graphdef?
> 
> The speedup provided by TensorRT varies based on the model and how you use TensorRT. For example the M40 is a fairly old card and so TensorRT doesn't necessarily optimize it as well as more recent cards; specifically the M40 doesn't have TensorCores and so TensorRT can use mixed-precision optimizations, which tend to give a large speedup.
> 
> Also, the TensorFlow model is using the GPU, so you are seeing acceleration of both models relative to CPU.
> 
> Finally, you should try a more recent version of TRTIS and TensorRT as both have improved significantly since 18.08. 19.03 was just released so that would be a good version to try.


According to your advice, I update TRTIS(tensorrtserver:19.02-py3) and TensorRT(TensorRT-5.0.2.6)，and the cuda version is 10, driver Version is 384 and GPU is Tesla P40, however, we run the inception-plan model in TRTIS, we got the error as:

E0405 11:57:25.732533 1 retrier.cc:37] Loading servable: {name: plan_inception version: 1} failed: Not found: output 'InceptionV3/Predictions/Softmax' not found for plan_inception_0_0_gpu0

The plan model input and output nodes are defined as:
parser.register_input("input", (3, 299, 299))
parser.register_output("InceptionV3/Predictions/Softmax")

In addition，the inception-graphdef model is run successfully in the same TRTIS:

I0405 11:56:43.065986 1 gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 22977 MB memory) -> physical GPU (device: 0, name: Tesla P40, pci bus id: 0000:04:00.0, compute capability: 6.1)
I0405 11:56:43.286886 1 loader_harness.cc:86] Successfully loaded servable version {name: inception_graphdef version: 1}

Could you give me some advices for that? Thank you.

**Comment by deadeyegoodwin** [comment]
For a TRT plan, the easiest thing to do is to run:
$ trtserver --strict-model-config=false --model-store=...

That will cause trtserver to create the config.pbtxt for you with the proper input and output names. You can then look at the actual config using the status API (curl localhost:8000/api/status). You can also use --log-verbose=1 to get more console output.

**Comment by deadeyegoodwin** [comment]
BTW, TRTIS console log should have indicated what the valid input/output tensor names were for the model. That is being fixed by https://github.com/NVIDIA/tensorrt-inference-server/pull/209

**Comment by ivankxt** [comment]
> For a TRT plan, the easiest thing to do is to run:
> $ trtserver --strict-model-config=false --model-store=...
> 
> That will cause trtserver to create the config.pbtxt for you with the proper input and output names. You can then look at the actual config using the status API (curl localhost:8000/api/status). You can also use --log-verbose=1 to get more console output.


According to your advice, I run the TRTIS as: 
$ trtserver  --strict-model-config=false --model-store=/data/nvidia_package/example/plan_inception

we got the error as:
I0406 02:43:30.668655 1 server.cc:761] Initializing TensorRT Inference Server
I0406 02:43:30.668727 1 server.cc:811] Reporting prometheus metrics on port 8002
I0406 02:43:30.669097 1 server.cc:1195] Starting server 'inference:0' listening on
I0406 02:43:30.669107 1 server.cc:1199]  localhost:8001 for gRPC requests
I0406 02:43:30.669199 1 server.cc:1103] Building nvrpc server
I0406 02:43:30.669216 1 server.cc:1109] Register TensorRT GRPCService
I0406 02:43:30.669226 1 server.cc:1112] Register Infer RPC
I0406 02:43:30.669230 1 server.cc:1116] Register Status RPC
I0406 02:43:30.669234 1 server.cc:1120] Register Profile RPC
I0406 02:43:30.669237 1 server.cc:1124] Register Health RPC
I0406 02:43:30.669240 1 server.cc:1128] Register Executor
I0406 02:43:30.671752 1 server.cc:1209]  localhost:8000 for HTTP requests
[evhttp_server.cc : 237] RAW: Entering the event loop ...
E0406 02:43:30.707766 1 server.cc:620] must specify platform for model '1'

In addition, model_repository is as below (Removed  the config.pbtxt file):
plan_inception/
├── 1
│ └── model.plan
└── inception_labels.txt

Could you give me some advices for that? Thank you.

**Comment by ivankxt** [comment]
> For a TRT plan, the easiest thing to do is to run:
> $ trtserver --strict-model-config=false --model-store=...
> 
> That will cause trtserver to create the config.pbtxt for you with the proper input and output names. You can then look at the actual config using the status API (curl localhost:8000/api/status). You can also use --log-verbose=1 to get more console output.

According to your advice, it worked. 
By the way, I want to know how to use  the low accuracy FP16 or INT8 in our plan model (Tesla P40 GPU and TensorRT-5.0.2.6) in order to have a much bigger speedup? 
Could you give me an example?
Thank you.

**Comment by deadeyegoodwin** [comment]
The TensorRT docs, specifically this section https://docs.nvidia.com/deeplearning/sdk/tensorrt-developer-guide/index.html#mixed_precision, describe how to use mixed-precision with TensorRT. It also points to several examples that you can look at.

**Comment by ivankxt** [comment]
> The TensorRT docs, specifically this section https://docs.nvidia.com/deeplearning/sdk/tensorrt-developer-guide/index.html#mixed_precision, describe how to use mixed-precision with TensorRT. It also points to several examples that you can look at.

According to the doc, I add the following two lines  in my python code to produce a new plan model and restarted the TRTIS, but we didn't get speedup. Could you give me some advices for that? Thank you.

builder.fp16_mode = True
builder.strict_type_constraints = True

`builder.fp16_mode = True
builder.strict_type_constraints = True`

**Comment by deadeyegoodwin** [comment]
The best place to get support for TensorRT questions like this is from the Developer Forum: https://devtalk.nvidia.com/default/board/304/tensorrt/
